addons/mrp/models/mrp_production.py

```diff
@@ -1864,98 +1857,132 @@ def _check_immediate(self):
+        for production, production_lot, (dummy, lot_id_quantity) in zip(productions, production_lots, lot_ids_quantities):
+            production.lot_producing_id = production_lot.id
+            production.product_qty = lot_id_quantity
             result.append((production.lot_producing_id.id, production.id))
-            production = backorder
+        # adapt remaining production
+        if not float_is_zero(remaining_quantity, precision_digits=precision_digits):
+            productions[-1].product_qty = remaining_quantity
+        # create moves
+        moves_vals = []
+        for production in productions:
+            if production.lot_producing_id:
+                production.qty_producing = production.product_qty
+            for index, move in enumerate(raw_moves):
+                new_move_uom_qty = float_round(raw_moves_quantities[index] * production.product_qty / quantity, precision_rounding=move.product_uom.rounding, rounding_method='HALF-UP')
+                if production == self:
+                    move.with_context(bypass_reservation_update=True).move_line_ids.product_uom_qty = 0
+                    move.with_context(do_not_unreserve=True).write({
+                        'product_uom_qty': new_move_uom_qty,
+                    })
+                else:
+                    moves_vals.append(move.copy_data({
+                        'raw_material_production_id': production.id,
+                        'product_uom_qty': new_move_uom_qty,
+                        'state': move.state,
+                        'reservation_date': move.reservation_date,
+                    })[0])
+            for index, move in enumerate(finished_moves):
+                new_move_uom_qty = float_round(finished_moves_quantities[index] * production.product_qty / quantity, precision_rounding=move.product_uom.rounding, rounding_method='HALF-UP')
+                if production == self:
+                    move.with_context(do_not_unreserve=True).write({
+                        'product_uom_qty': new_move_uom_qty,
+                    })
+                else:
+                    moves_vals.append(move.copy_data({
+                        'production_id': production.id,
+                        'product_uom_qty': new_move_uom_qty,
+                        'state': move.state,
+                        'reservation_date': move.reservation_date,
+                    })[0])
+        self.env['stock.move'].create(moves_vals)
+        # create movelines
+        movelines_vals = []
+        for production in productions:
+            for index, move in enumerate(production.move_raw_ids.filtered(lambda move: not move.additional)):
+                qty = move.product_uom_qty
+                while not float_is_zero(qty, precision_digits=precision_digits) and len(raw_movelines_quantities[index]) > 0:
+                    lot_id = raw_movelines_quantities[index][0][0]
+                    lot_qty = raw_movelines_quantities[index][0][1]
```

Comment:
Empty lists are falsy, and we can use destructuring assignment to assign two variables at once
```suggestion
                while not float_is_zero(qty, precision_digits=precision_digits) and raw_movelines_quantities[index]:
                    lot_id, lot_qty = raw_movelines_quantities[index][0]
```

Reply:
right

Reply:
code size reduced ;-)